### DISCUSSION TITLE
Player do not play all song in queue

### DISCUSSION THREAD

**Discussion by Domipoke** [Androz2091/discord-player#1795] [Q&A] [answered]
If I add 2 or more songs in queue it adds but the player stop playing finish the first one.
I try also adding QueueRepeatMode options but nothing works

My code:
```
const player = useMainPlayer();
const queue = player.nodes.create(guildID);
const result = await player.search(query);

// acquire task entry
const entry = queue.tasksQueue.acquire();
await entry.getTask();
queue.addTrack(result);

try {
    if (!queue.isPlaying()) await queue.node.play();
} finally {
    queue.tasksQueue.release();
}
```

**Comment by twlite** [comment, marked as answer]
Hello @Domipoke, based on the code snippet you've provided, it looks like you are not connecting to a channel. Please try connecting to a channel and see. In order to connect to a voice channel, use the following code snippet:

```js
if (!queue.channel) await queue.connect(voiceChannel);
```

It would be great if you could also share your debug logs. In order to enable debug logs, please use the following code snippet:

```js
client.once('ready', () => {
    const player = useMainPlayer();
    console.log(player.scanDeps());
    player.on('debug',console.log)
        .events.on('debug',(_,m) => console.log(m));
});
```

**Reply by Domipoke** [reply]
for now seems working, ty

**Comment by Domipoke** [comment]


**Reply by Domipoke** [reply]
this is the full code
